File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/Images2Api.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class Images2Api {
    private final HttpClient client;
    
    public Images2Api(HttpClient client) {
        this.client = client;
    }

    /** Create image variation */
    public ImageGenerationResponse createImageVariation(CreateImageVariationPostRequest body, Map<String, Object> params) throws Exception {
        return (ImageGenerationResponse) client.post(ApiPaths.aiPath("/v1/images/variations"), body, params, null, "multipart/form-data");
    }

    /** Create image */
    public ImageGenerationResponse createImage(ImageGenerationRequest body) throws Exception {
        return (ImageGenerationResponse) client.post(ApiPaths.aiPath("/v1/images/generations"), body);
    }

    /** Create image edit */
    public ImageGenerationResponse createImageEdit(CreateImageEditRequest body, Map<String, Object> params) throws Exception {
        return (ImageGenerationResponse) client.post(ApiPaths.aiPath("/v1/images/edits"), body, params, null, "multipart/form-data");
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/Images2Api.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class Images2Api(private val client: HttpClient) {

    /** Create image variation */
    suspend fun createImageVariation(body: CreateImageVariationPostRequest? = null, params: Map<String, Any>? = null): ImageGenerationResponse? {
        return client.post(ApiPaths.aiPath("/v1/images/variations"), body, params, null, "multipart/form-data") as? ImageGenerationResponse
    }

    /** Create image */
    suspend fun createImage(body: ImageGenerationRequest): ImageGenerationResponse? {
        return client.post(ApiPaths.aiPath("/v1/images/generations"), body) as? ImageGenerationResponse
    }

    /** Create image edit */
    suspend fun createImageEdit(body: CreateImageEditRequest? = null, params: Map<String, Any>? = null): ImageGenerationResponse? {
        return client.post(ApiPaths.aiPath("/v1/images/edits"), body, params, null, "multipart/form-data") as? ImageGenerationResponse
    }
}
